lib/raven/event.rb

```diff
@@ -159,14 +159,25 @@ def self.from_message(message, options = {})
     end
 
     def self.add_exception_interface(evt, exc)
-      evt.interface(:exception) do |int|
-        int.type = exc.class.to_s
-        int.value = exc.to_s
-        int.module = exc.class.to_s.split('::')[0...-1].join('::')
-
-        int.stacktrace = if exc.backtrace
-          StacktraceInterface.new do |stacktrace|
-            stacktrace_interface_from(stacktrace, evt, exc.backtrace)
+      evt.interface(:exception) do |exc_int|
+        exceptions = [exc]
+        while exc.respond_to?(:cause) && exc.cause
+          exceptions << exc.cause
+          exc = exc.cause
+        end
+        exceptions.reverse!
+
+        exc_int.values = exceptions.map do |exc|
+          SingleExceptionInterface.new do |int|
```

Comment:
Hmmm, I'm not really a fan of this name/a new class. Why can't you just use multiple instances of ExceptionInterface?